### PR TITLE
[Large 2D load for GEMM - #2] Support the repCluster field in convert DPAS layout from/to other layouts

### PR DESCRIPTION
**Pull request by chengjunlu**
Changes to the `emitOffset` for the DPAS layout to support `repCluster` field.

### CODE REVIEW

File: third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h [hunk 1]
```diff
@@ -148,17 +148,43 @@ static void
 emitOffsetForDpasLayoutPerCTA(const DpasEncodingAttr &dpasLayout,
                               SmallVector<SmallVector<unsigned>> &offsets,
                               unsigned ctaOffsetX, unsigned ctaOffsetY) {
+  // clang-format off
+  // For C operand the layout illustration.
```
Comment (etiotto) [inline third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h:152]: [readibility]:  For operand C the layout is:

File: third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h [hunk 2]
```diff
@@ -148,17 +148,43 @@ static void
...
+  //                      sub-group size 16
```
Comment (etiotto) [inline third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h:153]: I think we can remove subgroup size

File: third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h [hunk 3]
```diff
@@ -148,17 +148,43 @@ static void
...
+  ArrayRef<unsigned> repCluster = dpasLayout.getRepCluster();
+  SmallVector<unsigned> sizePerDPASInst = {sizePerThreads[0] / repCluster[0],
+                                           sizePerThreads[1] / repCluster[1]};
+
+  unsigned rowsPerElem = dpasLayout.getSubGroupSize() / instShapeC[1];
+  unsigned colsPerElem = 1;
+
+  unsigned repNumber = product<unsigned>(repCluster);
+  unsigned elemNumberPerRep = product<unsigned>(sizePerDPASInst);
+  for (unsigned repId = 0; repId < repNumber; repId++) {
```
Comment (etiotto) [inline third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h:174]: [minor]: use prefix increment operator to increment the induction variable.

**Review by etiotto — APPROVED** [review]
LGTM.

File: third_party/intel/lib/TritonIntelGPUToLLVM/ConvertLayoutOpToLLVM.cpp [hunk 4]
```diff
@@ -324,6 +302,20 @@ struct ConvertLayoutOpConversion
...
+    //    llvm::outs() << "johnlu layout:" << layout << "\n";
+    //    llvm::outs().flush();
+    //    llvm::outs() << "johnlu accumSizePerThread:" << accumSizePerThread <<
+    //    "\n"; llvm::outs().flush(); llvm::outs() << "johnlu the
+    //    numCTAsEachRep:"; llvm::outs().flush(); for (auto size :
+    //    numCTAsEachRep) {
+    //      llvm::outs() << size << ", ";
+    //    }
+    //    llvm::outs() << "\n";
+    //    llvm::outs().flush();
```
Comment (victor-eds) [inline third_party/intel/lib/TritonIntelGPUToLLVM/ConvertLayoutOpToLLVM.cpp:317]: Can we drop?

File: third_party/intel/lib/TritonIntelGPUToLLVM/ConvertLayoutOpToLLVM.cpp [hunk 5]
```diff
@@ -407,6 +399,8 @@ struct ConvertLayoutOpConversion
     auto dstShapePerCTATile = getShapePerCTATile(dstLayout, shape);
     auto shapePerCTA = getShapePerCTA(srcLayout, shape);
 
+    //    llvm::outs() << "johnlu the convert layout op: " << op << "\n";
+    //    llvm::outs().flush();
```
Comment (victor-eds) [inline third_party/intel/lib/TritonIntelGPUToLLVM/ConvertLayoutOpToLLVM.cpp:403]: Same

**Review by victor-eds — COMMENTED** [review]
LGTM but NIT

File: third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h (hunk 2 above)
Comment (chengjunlu) [reply third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h]: Done.

File: third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h (hunk 3 above)
Comment (chengjunlu) [reply third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h]: Done.

File: third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h (hunk 1 above)
Comment (chengjunlu) [reply third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h]: Done.

File: third_party/intel/lib/TritonIntelGPUToLLVM/ConvertLayoutOpToLLVM.cpp (hunk 4 above)
Comment (chengjunlu) [reply third_party/intel/lib/TritonIntelGPUToLLVM/ConvertLayoutOpToLLVM.cpp]: Done.

File: third_party/intel/lib/TritonIntelGPUToLLVM/ConvertLayoutOpToLLVM.cpp (hunk 5 above)
Comment (chengjunlu) [reply third_party/intel/lib/TritonIntelGPUToLLVM/ConvertLayoutOpToLLVM.cpp]: Done.

**Comment by chengjunlu** [comment]
> LGTM but NIT. Also, can we test?

Added some basic LIT test case.

**Review by etiotto — APPROVED** [review]

**Pull request merged**